dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/GridResponse.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.common;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author Jan Bernitt
+ */
+@Getter
+@AllArgsConstructor
+public final class GridResponse
+{
+    @JsonProperty
+    private final Pager pager;
+
+    @JsonProperty
+    private final Grid listGrid;
```

Comment:
Re the name `listGrid`, I think `grid` would be a better name, as the interface is called grid and `ListGrid` is just an implementation of it using an array list. Is it too bad of a breaking change to use `grid`? Has it always been `listGrid`?

Reply:
@larshelge I think `grid` would be a better name, too, but it's definitely easier on us if it isn't changed.  I estimate that it will take PEPFAR at least 20 hours of development time (and maybe several times that) to incorporate this into all of our DATIM and publicly-available apps.  I imagine other places with SQL view apps would have similar amounts of work.

Note that apps with SQL views that need to work on multiple versions of DHIS2 will need to have internal logic that could use either `listGrid` or `grid`.

I don't know what it was prior to DHIS 2.27, but I think it's been listGrid at least since then.

Reply:
Okay thanks Ben, agreed. Looking at the code I think the maintenance app also depends on that name, so let's leave it.